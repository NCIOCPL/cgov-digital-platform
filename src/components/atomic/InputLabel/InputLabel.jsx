import React from 'react';
import PropTypes from 'prop-types';

const InputLabel = ({
  label,
  labelHint,
  htmlFor,
  required,
  hasError,
  classes,
}) => {
  return (
    <label id={`${htmlFor}-label`} className={`${classes} ${required? '--required': ''}`} htmlFor={htmlFor}>
      {label}
      {labelHint && <span className="cts-input__hint"> {labelHint}</span>}
    </label>
  );
};

InputLabel.propTypes = {
  hasError: PropTypes.bool,
  htmlFor: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  labelHint: PropTypes.string,
  required: PropTypes.bool,
};

InputLabel.defaultProps = {
  sampleProperty: 'InputLabel',
};

export default InputLabel;
