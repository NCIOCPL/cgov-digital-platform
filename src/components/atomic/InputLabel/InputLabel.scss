.cts-label {
  display: inline-block;
  line-height: 1.5;
  max-width: 30rem;
  font-weight: 500;

  &--error {
    display: block;
    font-weight: bold;
    margin-top: 0;
  }
  &--required {
    color: #b50909;
  }
}

.cts-hint {
    color:#71767a;
    font-size: .75em;
}