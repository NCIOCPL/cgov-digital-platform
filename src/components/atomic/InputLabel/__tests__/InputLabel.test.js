import React from 'react';
import { shallow } from 'enzyme';
import InputLabel from '../InputLabel.jsx';

jest.unmock('../InputLabel.jsx');

describe('InputLabel', () => {

  let wrapper = null;
  const label = 'Label test';
  const htmlFor = 'htmlForTest';

  beforeEach(function() {
    wrapper = shallow(
      <InputLabel label={label} htmlFor={htmlFor} />
    );
  });

  it('is defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('renders the label text', () => {
    expect(wrapper.text()).toBe(label);
  });

  it('has an htmlFor attribute', () => {
    expect(wrapper.props().htmlFor).toBe(htmlFor);
  });

  it('has an id', () => {
    expect(wrapper.props().id).toBe(`${htmlFor}-label`);
  });

  it('accepts the required attribute', () => {
    wrapper = shallow(<InputLabel required label={label} htmlFor={htmlFor} />);
    expect(wrapper.hasClass('usa-label--required')).toBe(true);
  });
});
