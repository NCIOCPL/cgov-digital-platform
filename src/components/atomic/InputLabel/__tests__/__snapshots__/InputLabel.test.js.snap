// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel renders without error 1`] = `
<label
  className="cts-label"
  htmlFor="htmlForTest"
  id="htmlForTest-label"
>
  Label test
</label>
`;
